src/swaps/bitcoin.rs

```diff
@@ -1036,48 +1068,55 @@ impl BtcSwapTx {
 
             let mut witness = Witness::new();
             witness.push(final_schnorr_sig.to_vec());
+            refund_tx_cooperative.input[0].witness = witness;
```

Comment:
We need a witness for *all* inputs we want to spend

Reply:
Done: aff632f82e738b578eb618e7fce1ff91af78309d

One question I have here is this:

"Step 7: Get boltz's partial sig" starts by serializing the `refund_tx` to hex. It then calculates and attaches the witness for this input. This is inside a loop iterating over `refund_tx.input[input_index]`.

However this loop modifies the `refund_tx` at the end of each iteartion, by setting the witness for this input. So the tx hex will be different in each loop iteration. More specifically, it's this section: https://github.com/SatoshiPortal/boltz-rust/blob/e3f9c573b132eceef96465bbd811a1cbea0cfc61/src/swaps/bitcoin.rs#L995-L1060

I wonder, does the changing `refund_tx_hex` affect the partial sig calculations in Step 7, as they depend on this hex and are repeated for each input? Or does the partial sig logic account for this?

Reply:
It depends on what part of the transaction you are changing. Here we are just talking about the witness (= unlocking script here: https://wiki.bitcoinsv.io/index.php/SIGHASH_flags#Items_that_are_never_signed), which is not signed. You can ask for partial signatures of either the unsigned or partially signed transaction; both will work just fine